modules/contact/src/MechanicalContactConstraint.C

```diff
@@ -352,6 +356,36 @@ MechanicalContactConstraint::computeContactForce(PenetrationInfo * pinfo)
             pinfo->_mech_status=PenetrationInfo::MS_STICKING;
           break;
         }
+        case CF_TANGENTIAL_PENALTY:
+        {
+          // Frictional capacity (kinematic formulation)
+          const Real capacity( _friction_coefficient * (res_vec * pinfo->_normal > 0 ? res_vec * pinfo->_normal : 0) );
+
+          // Normal component of contact force (kinematic formulation)
+          RealVectorValue contact_force_normal( (-res_vec*pinfo->_normal) * pinfo->_normal );
+
+          // Predictor tangential component of contact force (penalty formulation)
+          RealVectorValue inc_pen_force_tangential = penalty * pinfo->_incremental_slip;
+          RealVectorValue contact_force_tangential = inc_pen_force_tangential + (pinfo->_contact_force_old - pinfo->_normal*(pinfo->_normal*pinfo->_contact_force_old));
+
+          // Magnitude of tangential predictor
+          const Real tan_mag( contact_force_tangential.norm() );
+
+          if (tan_mag > capacity)
+          {
+            pinfo->_contact_force = contact_force_normal + capacity * contact_force_tangential / tan_mag;
+            if (capacity == 0)
```

Comment:
This goes with the previous comment below. At first glance this looks like an integer comparison. I had to go up to look at the type of `_capacity`. Writing `0.0` instead of `0` makes it clearer to the reader that we are dealing with floats.